### PR TITLE
Add formatting options to DataGrid columns

### PR DESCRIPTION
**Pull request by Janpot**
Working on [the bundle size viewer](https://toolpad-pr-1449.onrender.com/deploy/clbnhjslx0001qd88yesm44v7/pages/c31adgx) use-case. Addressing some low-hanging fruits:

- Use `columnTypes` on the DataGrid
- Allow custom headerNames for DataGrid columns
- Allow picking a number format, starting with:
  - Plain
  - Percent
  - Currency
  - pretty print Bytes

For a preview: https://toolpad-pr-1449.onrender.com/deploy/clbnhjslx0001qd88yesm44v7/pages/c31adgx
The size and gzip columns have custom headers + formatting

https://user-images.githubusercontent.com/2109932/207665840-725c88f1-1c01-43ab-9b7d-9e64b049a0af.mov

### CODE REVIEW

File: packages/toolpad-app/src/toolpad/propertyControls/GridColumns.tsx [hunk 1]
```diff
@@ -15,29 +15,52 @@ import {
...
 // TODO: this import suggests leaky abstraction
 import { usePageEditorState } from '../AppEditor/PageEditor/PageEditorProvider';
 import { generateUniqueString } from '../../utils/strings';
 
+type GridAlignment = SerializableGridColumn['align'];
+
 const COLUMN_TYPES: string[] = ['string', 'number', 'date', 'dateTime', 'boolean', 'link', 'image'];
 const ALIGNMENTS: GridAlignment[] = ['left', 'right', 'center'];
 
+function formatOptionValue(numberFormat: NumberFormat | undefined) {
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpad/propertyControls/GridColumns.tsx:41]: what about calling it `formatNumberOptionValue`, or something else that makes it clearer that this applies to numbers?

File: packages/toolpad-app/src/toolpadDataSources/googleSheets/client.tsx [hunk 2]
```diff
@@ -204,7 +204,13 @@ function QueryEditor({
       {preview?.error ? (
         <ErrorAlert error={preview?.error} />
       ) : (
-        <DataGridPro sx={{ border: 'none' }} columns={columns} key={previewGridKey} rows={rows} />
+        <DataGridPro
+          sx={{ border: 'none' }}
+          columns={columns}
+          key={previewGridKey}
+          rows={rows}
+          columnTypes={CUSTOM_COLUMN_TYPES}
```
Comment (apedroferreira) [inline packages/toolpad-app/src/toolpadDataSources/googleSheets/client.tsx:212]: nice, i didn't know about this prop

**Review by apedroferreira — APPROVED** [review]
Looks good, I'm just not being able to authenticate in the link for some reason.
2 minor things I noticed from the video:

- ~~probably something like `$50.00` would be a better display than `US$50.00`?~~ (nevermind I guess it could be Canadian or Australian dollars and it comes from the library)
- the `plain` option could be preselected when the `number` type is picked

File: packages/toolpad-app/src/toolpadDataSources/googleSheets/client.tsx (hunk 2 above)
Comment (Janpot) [reply packages/toolpad-app/src/toolpadDataSources/googleSheets/client.tsx:212]: It's undocumented

**Review by bharatkashyap — APPROVED** [review]

**Pull request merged**